crates/edr_provider/src/data.rs

```diff
@@ -546,9 +542,9 @@ impl ProviderData {
     /// Estimate the gas cost of a transaction. Matches Hardhat behavior.
     pub fn estimate_gas(
         &self,
-        transaction: PendingTransaction,
+        transaction: ExecutableTransaction,
         block_spec: &BlockSpec,
-    ) -> Result<u64, ProviderError> {
+    ) -> Result<Result<u64, EstimateGasFailure>, ProviderError<LoggerErrorT>> {
```

Comment:
Same comment as here: https://github.com/NomicFoundation/hardhat/pull/4781/files#r1466504975

Reply:
See https://github.com/NomicFoundation/hardhat/pull/4781#discussion_r1465900562

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/4781/commits/c5a95d041b47d6e88ba5f5e0ba471ddb1d827df4